PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,11 @@
+#### What is the purpose of this pull request (PR)?
+
+#### Any background context to help the reviewer?
+
+#### Was this PR tested and how?
+
+#### Does this PR resolve an open issue?
```

Comment:
We usually put `Fixes #----` in the body of the pull to indicate that an open issue is resolved. Not sure if having a section for this in every pull makes sense.